tests/commands/test_store_commands.py

```diff
@@ -939,3 +941,257 @@ def test_status_with_multiple_branches(caplog, store_mock):
         "         beta/branch-2  15.0.0     15          2020-07-03T20:30:40+00:00",
     ]
     assert expected == [rec.message for rec in caplog.records]
+
+
+# -- tests for _get_lib_info helper
+
+def _create_lib(extra_content=None, metadata_id=None, metadata_api=None, metadata_patch=None):
+    """Helper to create the structures in disk for a given lib.
+
+    WARNING: this function has the capability of creating WRONG structures in disk.
+
+    This is specific for the _get_lib_info tests below, other tests should use the safe
+    similar one from factory.
+    """
+    base_dir = pathlib.Path('lib')
+    lib_file = base_dir / 'charms' / 'testcharm' / 'v3' / 'testlib.py'
+    lib_file.parent.mkdir(parents=True, exist_ok=True)
+
+    # save the content to that specific file under custom structure
+    if metadata_id is None:
+        metadata_id = "LIBID = 'test-lib-id'"
+    if metadata_api is None:
+        metadata_api = "LIBAPI = 3"
+    if metadata_patch is None:
+        metadata_patch = "LIBPATCH = 14"
+
+    fields = [metadata_id, metadata_api, metadata_patch]
+    with lib_file.open('wt', encoding='utf8') as fh:
+        for f in fields:
+            if f:
+                fh.write(f + '\n')
```

Comment:
side comment: interesting that @chipaca did not encourage `print(f, file=fh)` :-)

Reply:
I don't like to use `print` to write into files :)

Reply:
but everything is a file :sweat_smile: